### PR TITLE
Fix postgresql schema

### PR DESCRIPTION
**Pull request by medamineamara**
## What was changed

Add kubernetes job to support database schema migration and updates on PostgreSQL.

## Why?

Current workplace uses a Bring Your Own PostgreSQL and having this change upstream would be simple for more people.

## Checklist

1. Closes https://github.com/temporalio/helm-charts/issues/349

2. How was this tested:

Deployed Postgresql helm chart then deployed Temporal helm 
Complete logs :

```bash


amine@amine-Latitude-5420:~/Work/Temporal-helm-charts$ helm -n temporal ls
WARNING: Kubernetes configuration file is group-readable. This is insecure. Location: /home/amine/.kube/config
NAME      	NAMESPACE	REVISION	UPDATED                                	STATUS  	CHART            	APP VERSION
postgresql	temporal 	1       	2023-04-06 10:40:28.211826433 +0000 UTC	deployed	postgresql-12.1.6	15.1.0     


amine@amine-Latitude-5420:~/Work/Temporal-helm-charts$ helm -n temporal install -f temporal-values.yml temporal .
WARNING: Kubernetes configuration file is group-readable. This is insecure. Location: /home/amine/.kube/config
NAME: temporal
LAST DEPLOYED: Tue May 16 21:50:23 2023
NAMESPACE: temporal
STATUS: deployed
REVISION: 1
TEST SUITE: None
NOTES:
To verify that Temporal has started, run:

  kubectl --namespace=temporal get pods -l "app.kubernetes.io/instance=temporal"



amine@amine-Latitude-5420:~/Work/Temporal-helm-charts$ k -n temporal get pods
NAME                                   READY   STATUS      RESTARTS   AGE
postgresql-0                           1/1     Running     0          40d
temporal-admintools-6ff979d5fb-z5dwr   1/1     Running     0          98s
temporal-frontend-79fc54b7db-mknsd     1/1     Running     0          98s
temporal-history-dd6467bb7-jpt62       1/1     Running     0          98s
temporal-matching-5689b4b4f9-j4bsm     1/1     Running     0          98s
temporal-schema-setup-lzh9n            0/2     Completed   0          2m14s
temporal-schema-update-rlcrm           0/2     Completed   0          108s
temporal-web-fcbfff8c6-cvktl           1/1     Running     0          98s
temporal-worker-7b47ff8568-2xtdk       1/1     Running     0          98s


amine@amine-Latitude-5420:~/Work/Temporal-helm-charts$ helm -n temporal get values temporal
WARNING: Kubernetes configuration file is group-readable. This is insecure. Location: /home/amine/.kube/config
USER-SUPPLIED VALUES:
cassandra:
  enabled: false
debug: true
elasticsearch:
  enabled: false
grafana:
  enabled: false
postgresql:
  enabled: false
  image:
    pullPolicy: IfNotPresent
    repo: postgres
    tag: 15.1
prometheus:
  enabled: false
server:
  config:
    persistence:
      default:
        driver: sql
        sql:
          database: temporal
          driver: postgres
          host: postgresql.temporal.svc.cluster.local
          maxConnLifetime: 1h
          maxConns: 20
          password: ********** # Masked password
          port: 5432
          user: postgres
      visibility:
        driver: sql
        sql:
          database: temporal_visibility
          driver: postgres
          host: postgresql.temporal.svc.cluster.local
          maxConnLifetime: 1h
          maxConns: 20
          password: ********** # Masked password
          port: 5432
          user: postgres
  replicaCount: 1
web:
  image:
    tag: 2.12.0


amine@amine-Latitude-5420:~/Work/Temporal-helm-charts$ k logs temporal-schema-update-rlcrm -n temporal
Defaulted container "default-schema-update" out of: default-schema-update, visibility-schema-update, check-postgresql-service (init), check-postgresql (init)
2023-05-16T19:50:58.561Z	INFO	UpdateSchemeTask started	{"config": {"DBName":"","TargetVersion":"","SchemaDir":"./schema/postgresql/v96/temporal/versioned","IsDryRun":false}, "logging-call-at": "updatetask.go:98"}
2023-05-16T19:50:58.566Z	DEBUG	Schema Dirs: []	{"logging-call-at": "updatetask.go:187"}
2023-05-16T19:50:58.567Z	DEBUG	found zero updates from current version 1.9	{"logging-call-at": "updatetask.go:128"}
2023-05-16T19:50:58.567Z	INFO	UpdateSchemeTask done	{"logging-call-at": "updatetask.go:121"}


```

3. Any docs updates needed?

I think that in the readme the part where schema setup and update for postgresql should be removed
https://github.com/temporalio/helm-charts#install-with-your-own-postgresql

It's an extension of this PL https://github.com/temporalio/helm-charts/pull/350

### CODE REVIEW

File: values/values.postgresql.yaml [hunk 1]
```diff
@@ -49,6 +49,10 @@ mysql:
 
 postgresql:
   enabled: true
+  image:
```
Comment (brandonros) [inline values/values.postgresql.yaml:52]: is it possible to use this with an already existing postgres instance? the same way we are passing connection string properties like username/password/hostname/database name. i was trying to not stand up a separate postgresql just for temporal, but instead use a database inside an existing postgres instance hosted elsewhere

would these charts support this as is?

Comment (medamineamara) [reply values/values.postgresql.yaml:52]: yeah I'm already using it like that 

I'm not using the can `values/values.postgresql.yaml`

You can check my terminal logs

I have postgres chart already installed

Then I passed it in the values (check the values in hel get values)

Comment (medamineamara) [reply values/values.postgresql.yaml:52]: I'm going to update `values/values.postgresql.yaml` to avoid confusion

Comment (brandonros) [reply values/values.postgresql.yaml:52]: got it. so to clarify, i think the issue i had was, if i did `postgres: enabled: false`, i thought image wasn't required/was optional

it seems in the helm chart it's still required/not optional. i'm guessing that was my mistake, yes?

Comment (medamineamara) [reply values/values.postgresql.yaml:52]: yeah I think the postgresql image bloc is required to know which version to pull in the initContainers (for pg_isready tool)

image

**Comment by medamineamara** [comment]
I found a far more better approach 
https://github.com/temporalio/helm-charts/pull/281

**Comment by brandonros** [comment]
similar to https://github.com/temporalio/helm-charts/pull/350 and https://github.com/temporalio/helm-charts/pull/281 i'm pretty sure, probably https://github.com/temporalio/helm-charts/pull/275 too

**Pull request closed without merging**